src/gtfs/read.rs

```diff
@@ -199,12 +321,176 @@ pub fn read_stops<P: AsRef<path::Path>>(
+            name: "Métro".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Rail => objects::PhysicalMode {
+            id: "Train".to_string(),
+            name: "Train".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Bus => objects::PhysicalMode {
+            id: "Bus".to_string(),
+            name: "Bus".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Ferry => objects::PhysicalMode {
+            id: "Ferry".to_string(),
+            name: "Ferry".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::CableCar => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Gondola_SuspendedCableCar => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
+            co2_emission: None,
+        },
+        &RouteType::Funicular => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
+            co2_emission: None,
+        },
+        _ => objects::PhysicalMode {
+            id: "Bus".to_string(),
+            name: "Bus".to_string(),
+            co2_emission: None,
+        },
+    }
+}
+
+fn get_modes_from_gtfs(
+    gtfs_routes: &[Route],
+) -> (Vec<objects::CommercialMode>, Vec<objects::PhysicalMode>) {
+    let gtfs_mode_types: HashSet<RouteType> =
+        gtfs_routes.iter().map(|r| r.route_type.clone()).collect();
+
+    let commercial_modes = gtfs_mode_types
+        .iter()
+        .map(|mt| get_commercial_mode(mt))
+        .collect();
+    let physical_modes = gtfs_mode_types
+        .iter()
+        .map(|mt| get_physical_mode(mt))
+        .collect();
+    (commercial_modes, physical_modes)
+}
+
+fn get_lines_from_gtfs(gtfs_routes: &Vec<Route>, read_mode: RouteReadType) -> Vec<objects::Line> {
+    let mut lines = vec![];
+    match read_mode {
+        RouteReadType::RouteAsNtmLine => for r in gtfs_routes {
+            let line_code = match r.short_name.as_ref() {
+                "" => None,
+                _ => Some(r.short_name.to_string()),
+            };
+            let line_agency = match r.agency_id {
+                Some(ref agency_id) => agency_id.to_string(),
+                None => default_agency_id(),
+            };
+            let l = objects::Line {
+                id: r.id.to_string(),
+                code: line_code.clone(),
+                codes: vec![],
+                comment_links: vec![],
+                name: r.long_name.to_string(),
+                forward_name: None,
+                forward_direction: None,
+                backward_name: None,
+                backward_direction: None,
+                color: r.color.clone(),
+                text_color: r.text_color.clone(),
+                sort_order: None,
```

Comment:
I saw it with you, but I don't seem to be able to find it ! 
I checked https://developers.google.com/transit/gtfs/reference/ and http://gtfs.org/reference/ => no reference to it.
There is a mention here https://mbta.com/developers/gtfs-documentation but with an experimental status. I'm adding it just in case ...